public/js/home.js

```diff
@@ -1181,6 +1192,220 @@ function closeSideNav() {
+                timeoutFunction();
+                this.destroy();
+            }, options["timeout"]);
+        }
+
+        this.id;
+    }
+
+    /**
+     * creates the snackbar in the HTML
+     * if you want to show it as soon as you make it, use show without calling make instead
+     * returns the snackbar object
+     */
+    make() {
+        //stops if the element already exists
+        if (typeof document.getElementById(`sidenav-${this.id}`) === undefined) {
+            return;
+        }
+
+        //gives it an ID if it doesn't already have one
+        if (this.id === undefined) {
+            this.createID();
+        }
+
+        //creates the snackbar and gives it classes
+        const snackbarNode = document.createElement("DIV");
+        snackbarNode.classList.add("snackbar");
+        snackbarNode.classList.add("hidden");
+
+        //assigns it id based off of it's actual id
+        snackbarNode.id = `snackbar-${this.id}`;
+
+        //adds color if given
+        if (this.color !== undefined) {
+            snackbarNode.style.backgroundColor = `${this.color}`;
+        }
+
+        //sets the body onclick listener which just destroys it by default
+        const bodyOnClickFunction = this.bodyClick !== undefined ? () => {this.bodyClick();} : () => {};
+        const destroyFromBody = this.destroyWhenBodyClicked ? () => {this.destroy();} : () => {};
+        snackbarNode.addEventListener("click", () => {
+            bodyOnClickFunction();
+            destroyFromBody();
+        })
+
+        //adds the text
+        const textNode = document.createElement("SPAN");
+        textNode.textContent = this.text;
+
+        //colors the text if necessary
+        if (this.textColor !== undefined) {
+            textNode.style.color = this.textColor;
+        }
+
+        //adds the text node
+        snackbarNode.appendChild(textNode);
+
+        //makes the button if given button parameters
+        if (this.buttonText !== undefined && this.buttonClick != undefined) {
+
+            //creates the button and adds class
+            const buttonNode = document.createElement("BUTTON");
+
+            //creates the text span
+            const buttonTextNode = document.createElement("SPAN");
+            buttonTextNode.textContent = this.buttonText;
+
+            //colors the text if necessary
+            if (this.textColor !== undefined) {
+                buttonTextNode.style.color = this.textColor;
+            }
+
+            if (this.color !== undefined) {
+                buttonNode.style.backgroundColor = this.color;
+            }
+
+            //adds the text node
+            buttonNode.appendChild(buttonTextNode);
+
+            //sets the button onclick listener which runs the given funtion and destroys the snackar by default
+            const destroyFromButton = this.destroyWhenButtonClicked ? () => {this.destroy();} : () => {};
+            buttonNode.addEventListener("click", (event) => {
```

Comment:
Simplify arrow functions
```suggestion
            const destroyFromButton = this.destroyWhenButtonClicked ? () => this.destroy() : () => {};
            buttonNode.addEventListener("click", event => {
```